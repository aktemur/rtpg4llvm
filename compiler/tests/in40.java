/*compunit*/
extern int e;
double* d;
double g[5] = 5;
void foo() { }
int bar(int x, int y, double z) { return x + y * z; }
