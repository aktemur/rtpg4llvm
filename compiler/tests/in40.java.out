mkProgram(5, new ExternDecl(new IntType(), "e"),
   new GlobalDecl(new PtrType(new DoubleType()), "d"),
   new GlobalDecl(new ArrayType(new DoubleType(),5), "g", mkConst(5)),
   new FuncDeclaration(new VoidType(),
      "foo",
      mkFormals(0),
      mkScopedStmt(0)),
   new FuncDeclaration(new IntType(),
      "bar",
      mkFormals(3, new FuncParam(new IntType(), "x"),
         new FuncParam(new IntType(), "y"),
         new FuncParam(new DoubleType(), "z")),
      mkScopedStmt(1, new ReturnStmt(mkBinOp(Plus,
         new VarExpr("x"),
         mkBinOp(Star,
            new VarExpr("y"),
            new VarExpr("z")))))))
